core-web/src/main/java/org/silverpeas/core/web/treemenu/process/TreeBuilder.java

```diff
@@ -194,45 +148,172 @@ private static MenuItem buildFirstLevel(TreeFilter filter, String userId, String
+    if (TreeHandler.useOrder) {
+      nodeDetails.sort(Comparator.comparing(NodeDetail::getOrder));
+    }
+
+    for (NodeDetail nodeDetail : nodeDetails) {
+      // remove basket and declassified
+      if (nodeDetail.isBin() || nodeDetail.isUnclassified()) {
+        continue;
       }
+      MenuItem menuItem =
+          new MenuItem(Encode.forHtml(nodeDetail.getName(language)),
+              nodeDetail.getNodePK().getId(),
+              nodeDetail.getLevel(), NodeType.THEME, false, father, father.getKey());
+      menuItem.setNbObjects(nodeDetail.getNbObjects());
+      subElements.add(menuItem);
+    }
+    return subElements;
+  }
 
+  /**
+   * Get sub spaces
+   * @param father MenuItem parent
+   * @param userId the user identifier to display only the authorized item menu
+   * @param language the user language to display the label menu in the correct language
+   * @return a list of spaces MenuItem
+   */
+  private static List<MenuItem> getSubSpaces(MenuItem father, String userId, String language) {
+    List<MenuItem> subElements = new ArrayList<>();
+    List<SpaceInstLight> subspaces =
+        controller.getSubSpacesContainingComponent(father.getKey(), userId, KMELIA);
+    if (TreeHandler.useOrder) {
+      subspaces.sort(Comparator.comparing(SpaceInstLight::getOrderNum));
     }
-    return item;
+
+    for (SpaceInstLight space : subspaces) {
+      MenuItem item =
+          new MenuItem(space.getName(language), space.getId(), space.getLevel(),
+              NodeType.SPACE, false, father, null);
+      subElements.add(item);
+    }
+    return subElements;
+  }
+
+  /**
+   *
+   * @param father MenuItem parent
+   * @param userId the user identifier to display only the authorized item menu
+   * @param language the user language to display the label menu in the correct language
+   * @param filter determines what type of node and/or component must be display in the menu
+   * @return a list of components MenuItem
+   */
+  private static List<MenuItem> getSubComponents(MenuItem father, String userId, String language, TreeFilter filter) {
+    List<MenuItem> subElements = new ArrayList<>();
+    String[] componentIds = controller.getAvailCompoIdsAtRoot(father.getKey(),
+        userId);
+    int level = father.getLevel() + 1;
+    boolean isLeaf = false;
+    List<String> allowedComponents = filter.getComponents();
+
+    for (String componentId : componentIds) {
+      ComponentInstLight component = controller.getComponentInstLight(componentId);
+      // Default case : display all the component in the menu
+      if (allowedComponents.isEmpty() && filter.acceptNodeType(NodeType.COMPONENT)) {
+        isLeaf = getLeafValue(componentId);
+
+        org.silverpeas.core.web.treemenu.model.MenuItem item =
+            new org.silverpeas.core.web.treemenu.model.MenuItem(Encode.forHtml(component.getLabel(language)), componentId, level,
+                NodeType.COMPONENT, isLeaf, father, null);
```

Comment:
Why use here the package full namespace `org.silverpeas.core.web.treemenu.model.MenuItem` in this method instead of just the simple name 'MenuItem'?

Reply:
Weird.. fixed.